core/src/main/java/org/apache/iceberg/actions/RewriteFileGroup.java

```diff
@@ -31,26 +31,26 @@
 import org.apache.iceberg.util.DataFileSet;
 
 /**
- * Container class representing a set of files to be rewritten by a RewriteAction and the new files
- * which have been written by the action.
+ * Container class representing a set of data files to be rewritten by a RewriteAction and the new
+ * files which have been written by the action.
  */
-public class RewriteFileGroup {
-  private final FileGroupInfo info;
-  private final List<FileScanTask> fileScanTasks;
-
+public class RewriteFileGroup extends FileRewriteGroup<FileGroupInfo, FileScanTask, DataFile> {
   private DataFileSet addedFiles = DataFileSet.create();
 
+  /**
+   * @deprecated since 1.8.0, will be removed in 1.9.0.
+   */
+  @Deprecated
```

Comment:
Deprecated every object from the old API, but kept them for compatibility reasons.